scripts/gulpfiles/build_tasks.js

```diff
@@ -457,10 +467,19 @@ const pathSepRegExp = new RegExp(path.sep.replace(/\\/, '\\\\'), "g");
  */
 function flattenCorePaths(pathObject) {
   const dirs = pathObject.dirname.split(path.sep);
-  if (dirs[0] === 'core') {
-    pathObject.dirname = dirs[0];
-    pathObject.basename =
-        dirs.slice(1).concat(pathObject.basename).join('-slash-');
+
+  if (argv.compileTs) {
+    if (dirs[2] === 'core') {
+      pathObject.dirname = 'build/ts/core';
+      pathObject.basename =
+          dirs.slice(3).concat(pathObject.basename).join('-slash-');
+    }
+  } else {
+    if (dirs[0] === 'core') {
+      pathObject.dirname = dirs[0];
+      pathObject.basename =
+          dirs.slice(1).concat(pathObject.basename).join('-slash-');
+    }
```

Comment:
A shorter, more general version:
```suggestion
  const coreIndex = argv.compileTs ? 2 : 0;
  if (dirs[coreIndex] === 'core') {
    pathObject.dirname = path.join.apply(...dirs.slice(0, coreIndex + 1));
    pathObject.basename =
        dirs.slice(coreIndex + 1).concat(pathObject.basename).join('-slash-');
  }
```
Unfortunately neither version is totally safe (e.g. in the face of a hypothetical `blocks/x/core/` directory) but in practice this probably go away before any such directory gets created.